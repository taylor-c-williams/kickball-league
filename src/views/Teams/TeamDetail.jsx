import { useEffect, useState } from 'react';
import { Link, useParams, useHistory } from 'react-router-dom';
import { getRndInteger } from '../../services/global';
import { getTeamById, deleteTeamById } from '../../services/teams';
import RandomImg from '../../components/Players/RandomImg';

export default function TeamDetail({ min, max }) {
	const { id } = useParams();
	const [team, setTeam] = useState(null);
	const [loading, setLoading] = useState(true);
	const history = useHistory();

	useEffect(() => {
		getTeamById(id)
			.then((res) => setTeam(res))
			.finally(() => setLoading(false));
	}, [id]);

	if (loading) return <span className="loading">Loading...</span>;

	const handleDelete = async ({ id, name }) => {
		await deleteTeamById(id);
		history.push(`/teams/`);
	};

	return (
		<div className="content">
			<h1>{team.name}</h1>
			<RandomImg min={min} max={max} />
			{team.city}, {team.state}
			<ul>
				{team.players.map((player) => {
					return (
						<Link key={player.id} to={`/players/${player.id}`}>
							<li key={player.id}>
								{player.name} ◌ {player.position}
							</li>
						</Link>
					);
				})}
			</ul>
			<Link to={`/NewPlayer`}>Add Player</Link> ◌
			<Link to={`/editTeam/${team.id}`}>Update Team Details</Link> ◌{' '}
			<button
				type="button"
				aria-label={`Delete ${team.name}`}
				onClick={() => handleDelete({ id: team.id, name: team.name })}
			>
				Delete {team.name}
			</button>
		</div>
	);
}
