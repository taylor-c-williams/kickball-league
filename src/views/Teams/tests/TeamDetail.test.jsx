import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import TeamDetail from '../TeamDetail';

it('should render a detailed view of an individual team', async () => {
	const container = render(
		<MemoryRouter initialEntries={['/teams/1']}>
			<Route path="/teams/:id">
				<TeamDetail />
			</Route>
		</MemoryRouter>
	);

	screen.getByText('Loading...');

	const textEntry = await screen.findByText('Loading...', {
		exact: false,
	});

	expect(container).toMatchSnapshot();
	expect(textEntry).toBeInTheDocument();
});
