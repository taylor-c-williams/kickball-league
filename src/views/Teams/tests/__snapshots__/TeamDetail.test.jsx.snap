// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a detailed view of an individual team 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="content"
      >
        <h1>
          team name
        </h1>
        <section
          class="title-img"
        >
          <img
            alt="kickball"
            height="400"
            src="/kickballs/ball_14.jpg"
          />
        </section>
        team city
        , 
        team state
        <ul>
          <a
            href="/players/4"
          >
            <li>
              name player
               ◌ 
              position player
            </li>
          </a>
        </ul>
        <a
          href="/NewPlayer"
        >
          Add Player
        </a>
         ◌
        <a
          href="/editTeam/1"
        >
          Update Team Details
        </a>
         ◌
         
        <button
          aria-label="Delete team name"
          type="button"
        >
          Delete 
          team name
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="content"
    >
      <h1>
        team name
      </h1>
      <section
        class="title-img"
      >
        <img
          alt="kickball"
          height="400"
          src="/kickballs/ball_14.jpg"
        />
      </section>
      team city
      , 
      team state
      <ul>
        <a
          href="/players/4"
        >
          <li>
            name player
             ◌ 
            position player
          </li>
        </a>
      </ul>
      <a
        href="/NewPlayer"
      >
        Add Player
      </a>
       ◌
      <a
        href="/editTeam/1"
      >
        Update Team Details
      </a>
       ◌
       
      <button
        aria-label="Delete team name"
        type="button"
      >
        Delete 
        team name
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
